subprojects/docs/src/samples/userguide/java/basic/build.gradle

```diff
@@ -58,8 +58,10 @@ dependencies {
 }
 
 javadoc {
-    options.docletpath = configurations.asciidoclet.files.toList()
-    options.doclet = 'org.asciidoctor.Asciidoclet'
+    doFirst {
```

Comment:
Same as above.

Reply:
Then this is a bad example, isn't it? We should promote resolving configurations during the configuration phase.

`doFirst` is not good either, since it hides the inputs from up-to-date checks. The only thing which comes to mind how to fix this example is to use a configure task which configures the `javadoc` task at execution time. Do you have other ideas?